detective/functions.py

```diff
@@ -1,24 +1,60 @@
 """
 Helper functions.
 """
+import json
+import pandas as pd
 
+UNKNOWN = "unknown"
 
-def binary_state(value):
-    """Return a binary for the state of binary sensors"""
-    if value == "on":
-        return True
-    elif value == "off":
-        return False
-    else:
-        return float("nan")
-
-
-def isfloat(value):
-    """
-    Check if string can be parsed to a float.
-    """
-    try:
-        float(value)
-        return True
-    except ValueError:
-        return False
+
+def format_binary_state(state: str):
+    """Return a binary for the state of binary sensors."""
+    if state == "on":
+        return 1
+    elif state == "off":
+        return 0
+    return state
```

Comment:
could use an exception if binary is not ON/OFF

Reply:
Previously the binary version was returned as True/False, now as 0/1. I prefer the previous approach.

Reply:
I am not really sure about this one. Encoding as 0/1 made sense to me as I am basically interested in aggregating on counts. Probably there is a better way using pandas, perhaps [get_dummies](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.get_dummies.html) I will drop for now and give it some thought

Reply:
Or plain group_by/count. `get_dummies` looks similar to the [OneHotEncoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html) in scikit-learn.

Reply:
Encoding as 1/-1 would also make sense